File: Basic-Programs/Program20.java
import java.util.Scanner;

class Demo
{
    public float fPercentage = 0.0f;
    
    Demo(float fNo)
    {
        fPercentage = fNo;
    }
    
    void DisplayResult()
    {
        if((fPercentage >= 0.0f)  &&  (fPercentage <= 100.0f))        //  Filter 
        {
            if(fPercentage >= 85.0f)
            {
                System.out.println(" Grade A : Excellent ");
            }
            else if(fPercentage >= 75.0f)
            {
                System.out.println(" Grade B : Very Good ");
            }
            else if(fPercentage >= 60.0f)
            {
                System.out.println(" Grade C : Good ");
            }
            else if(fPercentage >= 40.0f)
            {
                System.out.println(" Grade D : Pass ");
            }
            else
            {
                System.out.println(" Fail ");
            }
        }
        else
        {
            System.out.println(" Invalid Percentage. please Enter the Value in Range 0 To 100 ");
            
        }
    }
}

class Program20
{
    public static void main(String A[])
    {
        float fValue = 0.0f;
        
        Scanner SC = new Scanner(System.in);
        
        System.out.println(" Please Enter Your Percentage : ");
        fValue = SC.nextFloat();
        
        Demo dobj = new Demo(fValue);
        dobj.DisplayResult();
        
    }
}

File: Basic-Programs/Program20.c
#include<stdio.h>

void DisplayResult(float fPercentage)
{
    if((fPercentage >= 0.0f) && (fPercentage <= 100.0f))        // Filter 
    {
        if(fPercentage >= 90.0f)
        {
            printf("Grade A : Excellent \n");
        }
        else if(fPercentage >= 75.0f)
        {
            printf("Grade B : Very Good \n");
        }
        else if(fPercentage >= 60.0f)
        {
            printf("Grade C : Good \n");
        }
        else if(fPercentage >= 40.0f)
        {
            printf("Grade D : Pass \n");
        }
        else
        {
            printf("Fail \n");
        }
    }
    else
    {
        printf("Invalid Percentage. Please Enter the Value in Range 0 To 100 \n");
    }
}

int main()
{
    float fResult = 0.0f;
    
    printf("Please Enter Your Percentage \n");
    scanf("%f",&fResult);
    
    DisplayResult(fResult);
    
    return 0;
}